app_core/gpio_commands.py

```
@@ -0,0 +1,220 @@
+def publish_pin_states(redis_client, states: List[Dict[str, Any]]) -> None:
+    """Publish a snapshot of GPIO pin states for the web UI to read."""
+    if redis_client is None:
+        return
+    try:
+        redis_client.set(
+            GPIO_PIN_STATE_KEY,
+            json.dumps({"pins": states, "ts": time.time()}),
+            ex=GPIO_PIN_STATE_TTL,
+        )
+    except Exception:
+        pass
```

Ruff (0.15.18)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)